Add tokens error

khorne3 opened the pull request:
<!---
Notes:
- Your PR should conform to our [Contributing Guidelines](https://github.com/auth0/docs/blob/master/CONTRIBUTING.md)
- If applicable, add details to the [update feed](https://github.com/auth0/docs/tree/master/updates)
- Make sure your PR gets deployed to a Heroku [Review App](https://github.com/auth0/docs/blob/master/CONTRIBUTING.md#review-apps) without errors.
- Please include a short description
- If your PR is a work in progress title it [DO NOT MERGE]
--->

Commit:
Add tokens error

luisrudge commented on articles/errors/libraries/auth0-js/invalid-token.md:
```diff
@@ -0,0 +1,23 @@
+---
+public: false
+---
+# HS256-Signed Tokens
```
Parsing a HS256-Signed id_token without an access_token

luisrudge commented on articles/errors/libraries/auth0-js/invalid-token.md:
```diff
@@ -0,0 +1,23 @@
+---
+# HS256-Signed Tokens
+
+Error: calls made to the **/userInfo** return the following:
```
Tokens signed with the HS256 algorithm cannot be properly validated. Instead, starting from auth0.js@v9 and lock@v11, a call to `/userinfo` is made with a valid `access_token`. If you don't have an access_token, you'll see the error:

luisrudge commented on articles/errors/libraries/auth0-js/invalid-token.md:
````diff
@@ -0,0 +1,23 @@
+accessToken parameter is not valid
+```
+
+## Cause
````
Not sure this section is needed, tbh. We already described the issue in the description above.

luisrudge commented on articles/errors/libraries/auth0-js/invalid-token.md:
```diff
@@ -0,0 +1,23 @@
+
+## Troubleshooting
+
+Change the value of your **responseType** parameter to **token id_token**. This will allow you to use an Access Token with the **/userInfo** endpoint.
```
- Change the value of your **responseType** parameter to **token id_token**, so you get an access_token in the response
- Change your client's hash algorithm to RS256. Go to https://manage.auth0.com/#/clients, select your client, click "Show advanced settings", click on OAuth and change the value of "JsonWebToken Signature Algorithm" to RS256

Commit:
Edit text

aaguiarz commented on articles/errors/libraries/auth0-js/invalid-token.md:
```diff
@@ -0,0 +1,29 @@
+
+Tokens signed with the HS256 algorithm cannot be properly validated.
+
+Beginning with **auth0.js version 9** and **Lock version 11**, API calls to **/userinfo** must be made with a valid Access Token.
```
This is not accurate, /userinfo always required an access token.

The issue here is that the user did not request an access_token. The fix implies requesting the token.

They don't need to change it RS256, but we can suggest them to do that. If they do, and they are using the access_token to call their own APIs, they will need to change the server code to validate the token with RS256 instead of HS256.

Commit:
Edit text

khorne3 commented:
@aaguiarz Do you suggest leaving the RS256 changes as is or separate it out as a "You might also consider..." tip?

aaguiarz committed:
Update invalid-token.md

- Rephrased the 'why this error occurred' section , and added more content to the advice regarding validating tokens.

aaguiarz commented:
@khorne3 Made some small changes, please review

Commit:
Update error message

Commit:
Update invalid-token.md

Commit:
Update invalid-token.md

woloski commented on articles/errors/libraries/auth0-js/invalid-token.md:
````diff
@@ -0,0 +1,33 @@
+```
+
+## Ways to fix this error
+
````
There are two ways to fix the error:

1) **(RECOMMENDED)** Change the client signature algorithm to RS256 instead of HS256.
2)  Change the value of your **responseType** parameter to **token id_token** (instead of the default), so that you receive an Access Token in the response.

To do (1) follow these instructions:
....
....

luisrudge committed:
Update invalid-token.md

luisrudge committed:
Update invalid-token.md

luisrudge committed:
Update invalid-token.md

luisrudge committed:
Update invalid-token.md

Commit:
Update invalid-token.md

woloski commented:
what are we waiting for?

khorne3 commented:
@woloski I was waiting for the go ahead. Will merge now.
